src/mono/wasm/runtime/exports.ts

```diff
@@ -199,7 +211,25 @@ function initializeImportsAndExports(
         warnWrap("addRunDependency", () => module.addRunDependency);
         warnWrap("removeRunDependency", () => module.removeRunDependency);
     }
+    let list: WeakRef<DotNetPublicAPI>[] = [];
+    if (!globalThisAny.getDotnetRuntime) {
+        globalThisAny.getDotnetRuntime = (runtimeId: number) => {
+            if (runtimeId < 0 || list.length < runtimeId) {
+                return undefined;
+            }
+            return list[runtimeId].deref();
+        };
+        globalThisAny.getDotnetRuntime.list = list;
+    }
+    else {
+        list = globalThisAny.getDotnetRuntime.list;
+    }
+    api.RuntimeId = globalThisAny.getDotnetRuntime.list.length;
+    list.push(create_weak_ref(api));
```

Comment:
since we're already defining getDotnetRuntime, i'd like a registerDotnetRuntime function instead of direct list manipulation, since I can imagine wanting to do work in there. It could have a __ prefix to indicate that it's for runtime use only